filestore/fs.py

```diff
@@ -214,33 +267,272 @@ def get_spec_handler(self, resource):
+
+
+
+        Parameters
+        ----------
+        resource_or_uid : Document or str
+            The resource to move the files of
+
+        new_root : str
+            The new 'root' to copy the files into
+
+        remove_origin : bool, optional (True)
+            If the source files should be removed
+
+        verify : bool, optional (False)
+            Verify that the move happened correctly.  This currently
+            is not implemented and will raise if ``verify == True``.
+
+        See Also
+        --------
+        `FileStore.shift_root`
+
+
+        .. Warning
+
+           This will change documents in your data base, move files
+           and possibly delete files.  Be sure you know what you are
+           doing.
+
+        '''
+        if self.version == 0:
+            raise NotImplementedError('V0 has no notion of root so can not '
+                                      'change it')
+        if verify:
+            raise NotImplementedError('Verification is not implemented yet')
+
+        datum_col = self._datum_col
+        # get list of files
+        actual_resource = self.resource_given_uid(resource_or_uid)
+        resource = dict(actual_resource)
+        resource.setdefault('root', '')
+        handler = self.get_spec_handler(resource['uid'])
+
+        datum_gen = self._api.get_datumkw_by_resuid_gen(datum_col,
+                                                        resource['uid'])
+        file_list = handler.get_file_list(datum_gen)
+
+        # check that all files share the same root
+        old_root = resource['root']
+        for f in file_list:
+            if not f.startswith(old_root):
+                raise RuntimeError('something is very wrong, the files '
+                                   'do not all share the same root, ABORT')
+
+        # sort out where new files should go
+        new_file_list = [os.path.join(new_root,
+                                      os.path.relpath(f, old_root))
+                         for f in file_list]
+        # copy the files to the new location
+        for fin, fout in zip(file_list, new_file_list):
+            # copy files
+            _make_sure_path_exists(os.path.dirname(fout))
+            shutil.copy2(fin, fout)
+
+        # update the database
+        new_resource = dict(resource)
+        new_resource['root'] = new_root
+
+        update_col = self._resource_update_col
+        resource_col = self._resource_col
+        ret = self._api.update_resource(update_col, resource_col,
+                                        old=actual_resource,
+                                        new=new_resource,
+                                        cmd_kwargs=dict(
+                                            remove_origin=remove_origin,
+                                            verify=verify,
+                                            new_root=new_root),
+                                        cmd='change_root')
+
```

Comment:
Some logging throughout this (potentially lengthy, definitely crucial) process might be a good idea. Or even `tqdm`... `change_root(..., tqmd_kwargs={'disable': False})` is always an option.